app/webpacker/controllers/toggle_controller.js

```diff
@@ -4,9 +4,22 @@ export default class extends Controller {
   static targets = ["content"];
 
   toggle(event) {
+    event.stopImmediatePropagation()
     const input = event.currentTarget;
+    const chevron = input.querySelector(".icon-chevron-down, .icon-chevron-up")
+    const toggleViaSingleElement = !input.dataset.toggleShow;
+
+    if(chevron) {
+      chevron.classList.toggle("icon-chevron-down");
+      chevron.classList.toggle("icon-chevron-up");
+    }
+
     this.contentTargets.forEach((t) => {
-      t.style.display = input.dataset.toggleShow === "true" ? "block" : "none";
+      if(toggleViaSingleElement) {
+        t.style.display = t.style.display === "none" ? "block" : "none";
+      } else {
+        t.style.display = input.dataset.toggleShow === "true" ? "block" : "none";
+      }
     });
```

Comment:
Hmm... it's not easy to read this bit of code and figure out what it's doing. It's works in two completely different ways in two different contexts. I had to read through the bits in the checkout controller and re-read this a few times before I understood the whole picture of what was happening. Which isn't ideal!

I'm wondering if we should maybe just split these two code paths into two separate functions? The original version toggles things _conditionally_ based on the `data-toggle-show` attribute and the new one toggles things unconditionally. They could be something like `toggle()` and `conditionalToggle()`?